src/sentry/notifications/manager.py

```diff
@@ -252,16 +253,25 @@ def get_for_recipient_by_parent(
         self,
         type: NotificationSettingTypes,
         parent: Union["Organization", "Project"],
-        recipient: "User",
+        recipient: Union["User", "Team"],
     ) -> QuerySet:
         """
         Find all of a project/organization's notification settings for a recipient.
         This will include each recipient's project/organization-independent settings.
         """
         scope_type = get_scope_type(type)
+        scope_identifier = parent.id
```

Comment:
You can replace the old `get_for_recipient_by_parent` and get_for_users_by_parent` with this:

```python
def get_for_recipient_by_parent(
    self,
    type_: NotificationSettingTypes,
    parent: Union["Organization", "Project"],
    recipients: Sequence[Union["Team", "User"]],
) -> QuerySet:
    from sentry.models import Team, User
    """
    Find all of a project/organization's notification settings for a list of
    users or teams. Note that this WILL work with a mixed list. This will
    include each user or team's project/organization-independent settings.
    """
    user_ids: MutableSet[int] = set()
    team_ids: MutableSet[int] = set()
    actor_ids: MutableSet[int] = set()
    for recipient in recipients:
        if type(recipient) == Team:
            team_ids.add(recipient.id)
        if type(recipient) == User:
            user_ids.add(recipient.id)
        actor_ids.add(recipient.actor.id)

    # If the list would be empty, don't bother querying.
    if not actor_ids:
        return self.none()

    parent_specific_scope_type = get_scope_type(type_)

    return self.filter(
        Q(
            scope_type=parent_specific_scope_type.value,
            scope_identifier=parent.id,
        ) | Q(
            scope_type=NotificationScopeType.USER.value,
            scope_identifier__in=user_ids,
        ) | Q(
            scope_type=NotificationScopeType.USER.value,
            scope_identifier__in=team_ids,
        ),
        type=type_.value,
        target__in=actor_ids,
    )
```
It should get both parent-independent and parent-specific settings for a project and set of recipients.